EvidenceApi/V1/Gateways/EvidenceGateway.cs

```diff
@@ -109,21 +110,41 @@ public List<EvidenceRequest> GetEvidenceRequests(ResidentSearchQuery request)
                 .ToList();
         }
 
-        public DocumentSubmissionQueryResponse GetPaginatedDocumentSubmissionsByResidentId(Guid id, int? limit = 10,
+        public DocumentSubmissionQueryResponse GetPaginatedDocumentSubmissionsByResidentId(Guid id, SubmissionState? state = null, int? limit = 10,
             int? page = 1)
         {
+            List<DocumentSubmission> documentSubmissions;
+            int total;
             var offset = (limit * page) - limit;
 
-            var total = _databaseContext.DocumentSubmissions
-                .Count(x => x.ResidentId.Equals(id));
-
-            var documentSubmissions = _databaseContext.DocumentSubmissions
-                .Where(x => x.ResidentId.Equals(id))
-                .Skip(offset ?? 0)
-                .Take(limit ?? 10)
-                .OrderByDescending(x => x.CreatedAt)
-                .ToList();
-
+            if (state != null)
+            {
+                documentSubmissions = _databaseContext.DocumentSubmissions
+                   .Where(x => x.ResidentId.Equals(id))
+                   .Where(x => x.State.Equals(state))
```

Comment:
These queries are executed lazily as long as you don't call `ToList`. That means you can build them up bit by bit in a conditional way, i.e. add the additional `Where` clause in an `if`, rather than having to write out two almost identical queries to get the evidence requests.

Reply:
Great suggestion, addressed in [bfe0a31](https://github.com/LBHackney-IT/evidence-api/pull/177/commits/bfe0a31d07aecda47afc9b200d1ee28936e692af)